File: solutions/python/resistor-color-trio/resistor_color_trio.py
import math

def format_ohms(ohm_value, label_metric_prefix):

    exp = 3 * ((len(str(ohm_value)) - 1) // 3)

    scaled = ohm_value / (10 ** exp)

    if scaled.is_integer():
        scaled_str = str(int(scaled))
    else:
        scaled_str = str(scaled)

    return f"{scaled_str} {label_metric_prefix}ohms"

def label(colors):

    color_codes = {
        "black"  : 0,
        "brown"  : 1,
        "red"    : 2,
        "orange" : 3,
        "yellow" : 4,
        "green"  : 5,
        "blue"   : 6,
        "violet" : 7,
        "grey"   : 8,
        "white"  : 9
    }

    ohm_value = (10 * color_codes[colors[0]] + color_codes[colors[1]]) * \
                pow(10, color_codes[colors[2]])

    label_metric_prefix = \
        ""     if len(str(ohm_value)) < 4 else  \
        "kilo" if len(str(ohm_value)) < 7 else  \
        "mega" if len(str(ohm_value)) < 10 else \
        "giga"

    return format_ohms(ohm_value, label_metric_prefix)

File: solutions/csharp/resistor-color-trio/ResistorColorTrio.cs
public static class ResistorColorTrio
{
    private static readonly Dictionary<string, int> ColorCodes =
        new Dictionary<string, int> {
            ["black"]  = 0,
            ["brown"]  = 1,
            ["red"]    = 2,
            ["orange"] = 3,
            ["yellow"] = 4,
            ["green"]  = 5,
            ["blue"]   = 6,
            ["violet"] = 7,
            ["grey"]   = 8,
            ["white"]  = 9
        };

    public static string Label(string[] colors)
    {
        long ohmValue =
            (10 * ColorCodes[colors[0]] + ColorCodes[colors[1]])
            * (long)Math.Pow(10, ColorCodes[colors[2]]);

        int digitCount = ohmValue.ToString().Length;

        string prefix =
            digitCount < 4  ? ""     :
            digitCount < 7  ? "kilo" :
            digitCount < 10 ? "mega" :
                              "giga";

        int scale = 3 * ((digitCount - 1) / 3);
        double scaled = ohmValue / Math.Pow(10, scale);

        string scaledStr = scaled.ToString().Replace(".0", "");

        return $"{scaledStr} {prefix}ohms";
    }
}
